### PR TITLE
Added closed event on server close

### PR DESCRIPTION
**Pull request by debjeet-sarkar**

### CODE REVIEW

**Comment by mcollina** [comment]
@debjeet-sarkar can you add a unit test? also can you rebase the commit? and add it to the README?

File: aedes.js [hunk 1]
```diff
@@ -283,6 +283,7 @@ function closeClient (client, cb) {
 Aedes.prototype.close = function (cb) {
   clearInterval(this._heartbeatInterval)
   clearInterval(this._clearWillInterval)
+  this.emit('closed')
   this._parallel(this, closeClient, Object.keys(this.clients), cb || noop)
```
Comment (mcollina) [inline aedes.js]: can you move the `emit` inside the `_parallel` callback.

Comment (debjeet-sarkar) [inline aedes.js]: Sure

File: aedes.js [hunk 2]
```diff
@@ -283,7 +283,16 @@ function closeClient (client, cb) {
 Aedes.prototype.close = function (cb) {
   clearInterval(this._heartbeatInterval)
   clearInterval(this._clearWillInterval)
-  this._parallel(this, closeClient, Object.keys(this.clients), cb || noop)
+  this._parallel(this, closeClient, Object.keys(this.clients), doneClose(this, cb))
```
Comment (mcollina) [inline aedes.js]: we should create a closure here, and call the callback when it's called.

**Pull request merged**